### ISSUE TITLE
Implement equivalent of Pandas to_sql dtype parameter

### ISSUE THREAD

**Issue by owen-synergy** [yehoshuadimarsky/bcpandas#8]
Thanks very much for this package, it definitely fills a need I have.

The only thing I'm missing is the equivalent of the pandas dtype parameter for passing a dictionary to set the column types.

Maybe an easy implementation would be to call the pandas to_sql method with an empty frame and the supplied dtype if the table doesn't already exist - if you think it's worth adding.

**Comment by yehoshuadimarsky** [comment]
Hmm interesting idea, wanna give it a try? You can open a pull request, let's work on it together and see if it's feasible

**Comment by owen-synergy** [comment]
Sure, I'll have a look. I think under the hood Pandas has another method called from to_sql to create the table if needed that maybe we can hook into directly to get equivalent functionality.

**Comment by owen-synergy** [comment]
Hi Josh, feel free to close this issue.

I've been losing my motivation to work on this given I've bumped into some of the limitations of the Pandas to_sql dtype parameter that mean I probably won't be using it in future. I'd been trying to have it setup SQL Server tables but there appears to be no way to set whether a column should be nullable or not or specify a primary key constraint - if those can't be specified in the pandas to_sql method I prefer to create the tables myself first.

I originally raised this issue because I thought I would allow the function to create the table and because I thought it would be nice to have more feature parity with the Pandas equivalent.

Thanks again for the awesome package though - once I've finished making my other changes I'll definitely benchmark it against to pandas to_sql and use bcpandas if the speed difference is significant for my dataset size.
